src/utils/Validate/validatePage.js

```diff
@@ -1,19 +1,22 @@
 // Local imports
 import validateComponent from './validateComponent';
+import showFormPage from '../FormPage/showFormPage';
 
 /**
  * Validate all of the components on a page.
+ * @param {object} page The page to validate, contains:
  * @param {Array} components The components to validate.
  * @param {object} formData The data to use that holds this components' values.
  * @returns An array containing all of the errors.
  */
-const validatePage = (components, formData) => {
-  if (Array.isArray(components)) {
-    return components.reduce((errors, component) => {
-      return errors.concat(validateComponent(component, formData));
+ const validatePage = (page) => {
+  if (Array.isArray(page?.components) && showFormPage(page, page.formData)) {
```

Comment:
If you flip these conditions around, you won't need the optional chaining as a `null` or `undefined` page won't be shown:
```suggestion
  if (showFormPage(page, page.formData) && Array.isArray(page.components)) {
```

Reply:
I made this change however the unit test now fails due to formData being null..

`  ● utils › Validate › page › should return no error when the components array is null

    TypeError: Cannot read properties of null (reading 'formData')

      11 |  */
      12 |  const validatePage = (page) => {
    > 13 |   if (showFormPage(page, page.formData) && Array.isArray(page.components)) {
         |                               ^
      14 |     return page.components.reduce((errors, component) => {
      15 |       return errors.concat(validateComponent(component, page.formData));
      16 |     }, []).filter(e => !!e).flat();

      at validatePage (src/utils/Validate/validatePage.js:13:31)
      at Object.<anonymous> (src/utils/Validate/validatePage.test.js:16:16)
`

I can avoid this by updated the first unit test to the following 
`      it('should return no error when the components array is null', () => {
        const PAGE = {
          components: null,
          formData: {
            alpha: 'alpha.smith@digital.homeoffice.gov.uk',
            bravo: 'bravo.jones@digital.homeoffice.gov.uk'
          }
        };
        expect(validatePage(PAGE).length).toEqual(0);
      });`
      
 Let me know if you would prefer that

Reply:
I have pushed a new commit addresses this and other comments - let me know if you would prefer to keep the optional chaining over editing the unit test and I can revert that part.